### ISSUE TITLE
[Low priority][Run 3] Tune Tracker Petal Wedge z difference error message to prevent confusion

### ISSUE THREAD

**Issue by cvuosalo** [cms-sw/cmssw#34476]
In Run 3 workflows 11634.0 and 11634.912, both DDD and DD4hep, an error message like this occurs in steps 2-5:
```
%MSG-e TkDetLayers:   TSGFromL2Muon:hltL3NoFiltersNoVtxTrajSeedOIHit@streamBeginRun  13-Jul-2021 21:57:44 CEST Run: 1 Stream: 0
 ForwardDiskSectorBuilderFromDet: Trying to build Petal Wedge from Dets at different z positions !! Delta_z = -0.951241
%MSG
```
This message comes from this code:
https://cmssdt.cern.ch/lxr/source/RecoTracker/TkDetLayers/src/ForwardDiskSectorBuilderFromDet.cc?v=CMSSW_10_6_20&%21v=CMSSW_12_0_X_2021-07-12-2300

The allowed difference is 0.8, and the value encountered is |0.95|. This message is more informational than an actual error at this point. If the difference were large, say > |2|, then that would be a real error.

A change could be made in the code to emit a Warning for absolute values from 0.8 to 2.0, and an Error for anything more.

Presently, the error can be confusing for developers who see the message and think their recent changes might have caused it, but this issue is only of low priority and maybe is not worth fixing.

**Comment by cvuosalo** [comment]
assign geometry,reco

**Comment by cvuosalo** [comment]
assign reco

**Comment by cvuosalo** [comment]
@adewit Does the error described in this issue look familiar to you? Do you have any ideas about it?

**Comment by mmusich** [comment]
@cvuosalo

>  I suspect an issue in the Tracker Reco geometry

It's not. The asymmetry of the endcap disks is a feature of the tracker misalignment scenario(-s) which has been present at various degrees throughout Run-2 and it is a reflection in the MC of the well known z-expansion effects in the Tracker Endcaps which recurrently plague the data alignment.
In my recollection we never experienced additional negative effects apart from the message, which is somewhat useful to track what geometries have it and what don't.
You can convince yourself that's the origin, because the workflows employing the not aligned geometry such as 12034.0 (`2021Design+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST`) do not emit this message, see e.g. [logs of the last IB ](https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_amd64_gcc900/CMSSW_12_0_X_2021-07-12-2300/pyRelValMatrixLogs/run/12034.0_TTbar_14TeV+2021Design+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST/step3_TTbar_14TeV+2021Design+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST.log)

**Comment by cvuosalo** [comment]
@mmusich The message is classified as Error. Should it be changed to Warning or Info? I think you are saying that this condition is not an actual error, so should the message classification be changed?

**Comment by mmusich** [comment]
I guess it could become a warning - though doing so - we might then hide more serious problems. 
Perhaps the threshold for emitting the error could be tuned, but honestly I wouldn't bother unless there is a strong push for that, as it's emitted only once per job.

**Comment by cvuosalo** [comment]
@mmusich Thanks for the information. It's good to know this error is not serious.

**Comment by cvuosalo** [comment]
+1

**Comment by cvuosalo** [comment]
I leave it to the Reco conveners whether to pursue this issue or close it.

**Comment by jpata** [comment]
+reconstruction
- given the above discussion, it seems fine to keep this occasional error about known tracker misalignment
